### PR TITLE
Simplify usage of suppress computed diff for Permissions

### PR DESCRIPTION
**Pull request by ethanzh**
- add a suppress-computed-diff directive to the entire `RequestConfig` object + other fields on the `RequestablePermission` object
- clean up `overlay.yaml` by organizing it into sections by Entity
- also add bulk requestable permissions data source
- clean up `README.md` and clarify `gen.yaml` version number

### CODE REVIEW

File: .speakeasy/gen.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ generation:
   auth:
     oAuth2ClientCredentialsEnabled: true
 terraform:
-  version: 0.11.0
+  version: 0.3.0
```
Comment (niamhdurfee) [inline .speakeasy/gen.yaml:15]: is this right?

File: internal/sdk/internal/hooks/registration.go [hunk 2]
```diff
@@ -12,7 +12,4 @@ func initHooks(h *Hooks) {
 	// Add hooks by calling h.register{SDKInit/BeforeRequest/AfterSuccess/AfterError}Hook
 	// with an instance of a hook that implements that specific Hook interface
 	// Hooks are registered per SDK instance, and are valid for the lifetime of the SDK instance
-	requestablePermissionsHook := &RequestablePermissionsHook{}
-
-	h.registerBeforeRequestHook(requestablePermissionsHook)
```
Comment (niamhdurfee) [inline internal/sdk/internal/hooks/registration.go:17]: add back?

File: .speakeasy/gen.yaml (hunk 1 above)
Comment (ethanzh) [reply .speakeasy/gen.yaml:15]: Yep!

File: internal/sdk/internal/hooks/registration.go (hunk 2 above)
Comment (ethanzh) [reply internal/sdk/internal/hooks/registration.go:17]: Yes, I guess the regeneration blows this away :(

File: internal/sdk/vendormanagement.go [hunk 3]
```diff
@@ -412,12 +412,12 @@ func (s *VendorManagement) CreateFoundDocumentMultipart(ctx context.Context, req
 
 }
 
-// CreateFoundDocumentJSON - Upload Found Documents
+// CreateFoundDocumentMultipart - Upload Found Documents
 // Upload a newly discovered document (e.g. Order Form, Quote) to Lumos for review
-func (s *VendorManagement) CreateFoundDocumentJSON(ctx context.Context, request shared.FoundDocumentsUpload, opts ...operations.Option) (*operations.CreateFoundDocumentJSONResponse, error) {
+func (s *VendorManagement) CreateFoundDocumentMultipart(ctx context.Context, request operations.CreateFoundDocumentMultipartRequestBody, opts ...operations.Option) (*operations.CreateFoundDocumentMultipartResponse, error) {
```
Comment (niamhdurfee) [inline internal/sdk/vendormanagement.go:417]: where is this multipart thing coming from? I vaguely recall removing it from the API?

**Review by niamhdurfee — CHANGES_REQUESTED** [review]
I think the hooks registration needs to be replaced but that's about all I noticed!

File: internal/sdk/vendormanagement.go (hunk 3 above)
Comment (ethanzh) [reply internal/sdk/vendormanagement.go:417]: This part is black magic to me to be honest. Part of the downside of relying on a generated provider -- it's less explainable to see what this change actually means

File: internal/sdk/internal/hooks/registration.go (hunk 2 above)
Comment (ethanzh) [reply internal/sdk/internal/hooks/registration.go]: Re-added

**Pull request merged**